schemaregistry/schemaregistry.md

```diff
@@ -0,0 +1,257 @@
+transparency about which `id` have already been taken. An implementation MAY
+withhold all details other than the `id` attribute from a user not authorized
+to access the schema group.
+
+This specification does not define management constructs for such access control
+rules.
+
+### 2.1.1. Schema group attributes
+
+The data model for a schema group consists of three attributes:
+
+#### id
+
+- Type: `String`
+- Description: Identifies the schema group.
+- Constraints:
+  - REQUIRED
+  - MUST be a non-empty string
+  - MUST conform with RFC3986/3.3 `segment-nz-nc` syntax
+  - MUST be unique within the scope of the schema registry
+- Examples:
+  - mygroup
+  - my-group
+
+#### format
+
+- Type: `String`
+- Description: Defines the schema format managed by this schema group. The
+  formats supported by an implementation are specific to the implementation.
+  This specification does not mandate support for particular formats.
+- Constraints:
+  - OPTIONAL. If not set, each schema in the group MAY have multiple, concurrent
+    format representations. See XXXX
+  - MUST be a non-empty string, if present.
+- Examples:
+  - "avro"
+  - "json-schema"
+  - "xsd11"
+
+#### description
+
+- Type: `String`
+- Description: Explains the purpose of the schema group.
+- Constraints:
+  - OPTIONAL
+- Examples:
+  - "This group holds schemas for the fabulous example app."
+
+## 2.2. Schema and Schema Version
+
+Conceptually, a schema is a description of a data structure. Since data
+structures evolve over time, the schema describing them will also evolve over
+time. Therefore, a schema often has multiple versions.
+
+For simple scenarios, the API allows for version management to be automatic and
+transparent. Whenever a schema is updated, a new version number is assigned and
+prior schema versions are retained. The latest available schema is always the
+default version that is retrieved when the API is given just the schema `id`.
+
+A newer schema version might introduce breaking changes or it might only
+introduce careful changes that preserve compatibility. These strategies are not
+subject of this specification, but the API provides a conflict handling
+mechanism that allows an implementation to reject updates that do not comply
+with a compatibility policy, if one has been implemented.
+
+### 2.2.1. Multi-format schema sets
+
+If the schema format has not been restricted at the schema group level, each
+schema may describe the same data structure for encoding in different formats
+concurrently. That implies that the schema managed by this registry is not
+a single document, but MAY be a set of documents that all describe the exact
+same data structure, each in their own particular syntax and using their own 
+type model.
+
+The schema version therefore relates to this abstract notion of schema and the
+described data structure. All documents coexisting within the same version
+SHOULD describe the exact same data structure.
+
+### 2.2.2. Schema attributes
```

Comment:
It would be useful to keep some other meta-data for each artifact, particularly if anyone eve wants to create a Registry UI.  Name, description, labels, creationTime, modifiedTime, etc.